### PR TITLE
Only handle pod on pod labels, annotation or phase update

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/ambientindex.go [hunk 1]
```diff
@@ -777,14 +787,14 @@ func (a *AmbientIndex) handleService(obj any, isDelete bool, c *Controller) map[
...
-		wl := a.byPod[p.Status.PodIP]
 		// Can be nil if it's not ready, hostNetwork, etc
+		wl := c.extractWorkload(p)
 		if wl != nil {
-			wl = c.extractWorkload(p)
 			// Update the pod, since it now has new VIP info
 			a.byPod[p.Status.PodIP] = wl
 			wls = append(wls, wl)
 		}
+
```
Comment (hanxiaop) [inline pilot/pkg/serviceregistry/kube/controller/ambientindex.go:797]: extra line

File: pilot/pkg/serviceregistry/kube/controller/ambientindex.go [hunk 2]
```diff
@@ -691,11 +693,19 @@ func (c *Controller) setupIndex() *AmbientIndex {
 	return &idx
 }
 
-func (a *AmbientIndex) handlePod(_, newObj any, isDelete bool, c *Controller) sets.Set[model.ConfigKey] {
+func (a *AmbientIndex) handlePod(oldObj, newObj any, isDelete bool, c *Controller) sets.Set[model.ConfigKey] {
+	p := controllers.Extract[*v1.Pod](newObj)
+	old := controllers.Extract[*v1.Pod](oldObj)
+	if old != nil {
+		// compare only labels and pod phase, which are what we care about
+		if reflect.DeepEqual(old.Labels, p.Labels) && old.Status.Phase == p.Status.Phase &&
```
Comment (howardjohn) [inline pilot/pkg/serviceregistry/kube/controller/ambientindex.go:701]: maps.Equal is preferred, faster than reflect

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/ambientindex.go:701]: pod IP is allocated before pod running, before that pod status won't be ready and phase not be running

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/ambientindex.go:701]: So that means after ip allocated the phase must change

**Review by howardjohn — APPROVED** [review]

**Pull request merged**